add a fips option to all existing commands (#1501)
DataDog/datadog-ci: README.md

#### `gate`
- `evaluate`: Evaluate [Quality Gates](src/commands/gate) rules in Datadog. [📚](https://docs.datadoghq.com/quality_gates/)

### FIPS support

> The `fips` option allow `datadog-ci` to use a FIPS cryptographic module provider, if the OpenSSL library installed on the host system provides it.

Reviewer: ```suggestion
The `fips` option allows `datadog-ci` to use a FIPS cryptographic module provider, if the OpenSSL library installed on the host system provides it.
```

> `datadog-ci` cannot assert if such a provider is available, and won't throw any error in case the provider is not FIPS compliant.

Reviewer: ```suggestion
Note that `datadog-ci` cannot assert if such a provider is available, and won't throw any error in case the provider is not FIPS compliant.
```

Reviewer: > in case the provider is not FIPS compliant

Did you mean FIPS validated?

Author: Yes, thanks for catching that! 🙇

> However, node version below 17 are incompatible with OpenSSL 3 which provides FIPS support.
> In this case, enabling the `fips` option will cause the command to throw an error.
> The option `fips-ignore-error` is meant to ignore this error.

Reviewer: ```suggestion
Node.js versions below 17 are incompatible with OpenSSL 3 which provides FIPS support.
In this case, enabling the `fips` option will cause the command to throw an error.
The option `fips-ignore-error` is meant to ignore this error.
```

#### `fips`
Enable FIPS support by datadog-ci, given a FIPS provider is installed on the host system.
If no FIPS provider is installed, no error are raised.

ENV variable: DATADOG_FIPS=True
CLI param: `--fips`

#### `fips-ignore-error`

> Ignore node errors in case FIPS cannot be enabled on the host system.

Reviewer: ```suggestion
Ignore Node.js errors in case FIPS cannot be enabled on the host system.
```

### FIPS support

The `fips` option allows `datadog-ci` to use a FIPS cryptographic module provider if the OpenSSL library installed on the host system provides it.

**Note**: `datadog-ci` cannot assert if such a provider is available, and doesn't throw any error if the provider is not FIPS validated.

Node.js versions below 17 are incompatible with OpenSSL 3, which provides FIPS support.

> If you are using a Node.js version below 17, enabling the `fips` option will cause the command to throw an error.

Reviewer 2: ```suggestion
If you are using a Node.js version below 17, enabling the `fips` option causes the command to throw an error.
```

Author: Thanks for catching that! 🙇
